src/cli/tools/java/index.ts

```diff
@@ -0,0 +1,182 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { execa } from 'execa';
+import { inject, injectable } from 'inversify';
+import semver from 'semver';
+import { InstallToolBaseService } from '../../install-tool/install-tool-base.service';
+import { PrepareToolBaseService } from '../../prepare-tool/prepare-tool-base.service';
+import {
+  CompressionService,
+  EnvService,
+  HttpService,
+  PathService,
+} from '../../services';
+import { logger } from '../../utils';
+import type { JavaReleases } from './schema';
+
+const baseUrl =
+  'https://github.com/joschi/java-metadata/raw/main/docs/metadata/ga/linux';
+const fileUrl = 'hotspot/temurin.json';
```

Comment:
@rarkins this is new new source. Should we continue to use `api.adoptium.net` instead?

Reply:
Does it have the same versioning syntax as adoptium? i.e. so that every "install-tool x.y.z+foo" in Dockerfiles today will continue to work?

And will the list of versions be consistent with what adoptium returns? We probably should update Renovate to match sources so that we reduce the chance of a future mismatch

Reply:
yes, this file is created from the adoptium api